src/app/home/home.component.ts

```diff
@@ -25,4 +45,35 @@ export class HomeComponent implements CanComponentDeactivate{
     return true; // Allow navigation if there are no unsaved changes
   }
 
+  // Edit a user
+  onEdit(user: any) {
+    const updatedUserData = { name: 'Updated Name', email: 'updated@example.com' };
+    this.api.editUser(user.id, updatedUserData).subscribe(
+      (response) => {
+        console.log('User updated:', response);
+        // Update the local users array with the new data
+        const index = this.users.findIndex(u => u.id === user.id);
+        if (index !== -1) {
+          this.users[index] = response;
+        }
+      },
+      (err) => {
+        console.log('Error updating user:', err);
+      }
+    );
+  }
+
+  // Delete a user
+  onDelete(user: any) {
+    this.api.deleteUser(user.id).subscribe(
+      (response) => {
+        console.log('User deleted:', response);
+        // Remove the user from the local users array
+        this.users = this.users.filter(u => u.id !== user.id);
```

Comment:
It is okay, but it would be better to call the fetch API again if any other user data was updated.